modules/marsmediaBidAdapter.js

```diff
@@ -295,6 +316,126 @@ function MarsmediaAdapter() {
 
     return bids;
   };
+
+  function sendbeacon(bid, type) {
+    const bidString = JSON.stringify(bid);
+    const encodedBuf = window.btoa(bidString);
+    utils.triggerPixel('https://ping-hqx-1.go2speed.media/notification/rtb/beacon/?bt=' + type + '&bid=3mhdom&hb_j=' + encodedBuf, null);
+  }
+
+  /**
+   * Gets bidfloor
+   * @param {Object} bid
+   * @returns {Number} floor
+   */
+  function _getFloor (bid) {
+    const curMediaType = bid.mediaTypes.video ? 'video' : 'banner';
+    let floor = 0;
+
+    if (typeof bid.getFloor === 'function') {
+      const floorInfo = bid.getFloor({
+        currency: 'USD',
+        mediaType: curMediaType,
+        size: '*'
+      });
+
+      if (typeof floorInfo === 'object' &&
+        floorInfo.currency === 'USD' &&
+        !isNaN(parseFloat(floorInfo.floor))) {
+        floor = floorInfo.floor;
+      }
+    }
+
+    return floor;
+  }
+
+  function _getMinSize(sizes) {
+    return sizes.reduce((min, size) => size.h * size.w < min.h * min.w ? size : min);
+  }
+
+  function _isViewabilityMeasurable(element) {
+    return !_isIframe() && element !== null;
+  }
+
+  function _isIframe() {
+    try {
+      return utils.getWindowSelf() !== utils.getWindowTop();
+    } catch (e) {
+      return true;
+    }
+  }
+
+  function _getViewability(element, topWin, { w, h } = {}) {
+    return utils.getWindowTop().document.visibilityState === 'visible'
```

Comment:
Can this be `topWin.document.visibilityState`? Does that `utils` function return something different?